spring-cloud-starter-aws-parameter-store-config/src/main/java/io/awspring/cloud/autoconfigure/paramstore/AwsParamStoreConfigDataLocationResolver.java

```diff
@@ -103,8 +104,11 @@ public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDat
 		registerBean(context, type, supplier);
 		context.getBootstrapContext().addCloseListener(event -> {
 			T instance = event.getBootstrapContext().get(type);
-			event.getApplicationContext().getBeanFactory().registerSingleton("configData" + type.getSimpleName(),
-					instance);
+			String name = "configData" + type.getSimpleName();
+			ConfigurableApplicationContext appContext = event.getApplicationContext();
```

Comment:
Please add a comment here and in corresponding Secrets Manager class explaining why this check is necessary. It won't take too much time until it is forgotten.